### ISSUE TITLE
Support for DDB parallel scans

### ISSUE THREAD

**Issue by googley42** [zio/zio-dynamodb#43]
[DDB has support for parallel scans](https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/Scan.html#Scan.ParallelScan)
1) I think the main work here would be in the `DynamoDBExeceutor` implementation 
2) some sort of flag could be passed from the query DSL to indicate that a parallel scan is required

**Comment by googley42** [comment]
so I remembered the loose idea I had when I wrote the parallel scan issue - whether it is actually possible I am not sure. What I thought was that the DSL passed in the number of segments N (where N > 1 ) when a parallel scan is required. The DynamoDBExecutorImpl then divided this up into N parallel scan queries and managed the segment number. Ideally each query is executed in a separate fiber and the separate results streams (which would also manage any pagination if required) are merged back into one stream as the DSL return value. So to the user of the DSL a parallel query is used just like a normal query but with an extra N segments param. We could restrict parallel queries to just ScanAll/ScanAllItem methods as these return ZStreams. As mention above, whether it is actually possible I am not sure.
